src/Application/SaveMappings/SaveMappingsPresenter.php

```diff
@@ -17,4 +19,6 @@ public function presentSaveFailed(): void;
 
 	public function presentInvalidEntityId(): void;
 
+	public function presentPermissionDenied( Throwable $exception ): void;
```

Comment:
This is a more specific instance of "save failed" and it seems useful to provide the upstream error message in the response.